include/znc/Modules.h

```diff
@@ -720,6 +720,10 @@ class CModule {
 	 *  @return See CModule::EModRet.
 	 */
 	virtual EModRet OnUserTopicRequest(CString& sChannel);
+	/** This module hook is called when a user requests to quit from network.
+	 *  @param sMessage The quit message the client sent.
+	 */
+	virtual void OnUserQuit(CString& sMessage);
```

Comment:
Some module might want to abort the disconnect, and not make client to disconnect.